### PR TITLE
[WIP] Further development on intel_mp driver

### PR DESCRIPTION
**Pull request by takikawa**
The initial work on the `intel_mp` driver is now pending for some upstream review. This PR is for tracking additional work on the driver (keeping it on a separate branch so that the initial PR doesn't just keep growing) and making sure that CI passes.

Progress on this branch:
 - [x] pool mirroring for VMDq
 - [x] basic performance improvements for `intel_mp`
 - [x] improvements for debugging
 - [x] support for setting counters for queues (Tx and Rx)
 - [x] ported priority/rate limiting code from intel10g
 - [x] fixed some buggy app teardown code
 - [ ] more performance improvements
 - [x] make rx/tx queue numbers relative to the selected pool number (on other branch)

### CODE REVIEW

File: src/apps/intel_mp/intel_mp.lua [hunk 1]
```diff
@@ -1047,14 +1047,13 @@ function Intel1g:init ()
 end
 
 function Intel1g:link_status ()
-   local mask = bits { Link_up = 1 }
-   return bit.band(self.r.STATUS(), mask) == mask
+   return bit.band(self.r.STATUS(), lshift(1, 1)) == mask
```
Comment (teknico) [inline src/apps/intel_mp/intel_mp.lua]: On the right side you're still using `mask`, which is not defined anymore.

Comment (takikawa) [reply src/apps/intel_mp/intel_mp.lua]: Whoops, nice catch. This is the problem with code that you can't test. :p

File: src/apps/intel_mp/intel_mp.lua [hunk 2]
```diff
@@ -324,7 +326,9 @@ function Intel:new (conf)
       vlan = conf.vlan,
       want_mirror = conf.mirror,
       rxcounter = conf.rxcounter,
-      txcounter = conf.txcounter
+      txcounter = conf.txcounter,
+      limit = conf.rate_limit,
```
Comment (teknico) [inline src/apps/intel_mp/intel_mp.lua]: I'd rather keep calling this `rate_limit`, because just `limit` is unclear. What do you think?

Comment (takikawa) [reply src/apps/intel_mp/intel_mp.lua]: I like your suggestion, I'll change it.

**Comment by takikawa** [comment]
I'm going to close this since we are no longer doing development on this branch (deleted the branch too, but I took a backup).

**Pull request closed without merging**